rpa-mapper/src/main/java/uk/gov/hmcts/cmc/rpa/mapper/DefenceResponseJsonMapper.java

```diff
@@ -26,17 +35,59 @@ public DefenceResponseJsonMapper(DefendantJsonMapper defendantMapper) {
     }
 
     public JsonObject map(Claim claim) {
-        Response response = claim.getResponse().orElseThrow(IllegalStateException::new);
+        Response response = claim.getResponse().orElseThrow(IllegalArgumentException::new);
         String defendantsEmail = claim.getDefendantEmail();
+
         return new NullAwareJsonObjectBuilder()
             .add("caseNumber", claim.getReferenceNumber())
             .add("responseSubmittedOn", DateFormatter.format(claim.getRespondedAt()))
             .add("defenceResponse", defenceResponse(response))
             .add("defendant", defendantMapper.map(response.getDefendant(), claim.getClaimData().getDefendant(), defendantsEmail))
             .add("mediation", isMediationSelected(response))
+            .add("amountAdmitted", getAmountAdmitted(response))
+            .add("payment", mapPayment(response))
             .build();
     }
 
+    private BigDecimal getAmountAdmitted(Response response) {
+        if (response instanceof PartAdmissionResponse) {
+            return ((PartAdmissionResponse) response).getAmount();
+        }
+        return null;
+    }
+
+    private JsonObject mapPayment(Response response) {
+        JsonObjectBuilder jsonObjectBuilder = new NullAwareJsonObjectBuilder();
+        if (ResponseUtils.isAdmissionResponse(response)) {
+            updateJsonBuilderForAdmissions(jsonObjectBuilder, response);
+            return jsonObjectBuilder.build();
+        }
+
+        return null;
+    }
+
+    private void updateJsonBuilderForAdmissions(JsonObjectBuilder jsonObjectBuilder, Response response) {
+        PaymentIntention paymentIntention = null;
+        if (response instanceof PartAdmissionResponse) {
```

Comment:
comment: i dont like use of `instanceof` - afaik its bad practice. it looks like we use it quite a bit though. just makes extending harder - if we add another PaymentIntention type we need to be careful where we implement logic like this. i will approve though, maybe we look at as a refactoring task one day.